bin/load-module-esm-support.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+var fs = require('fs');
+var path = require('path');
+var dirnamePath = path.dirname;
+var extnamePath = path.extname;
+var resolvePath = path.resolve;
+
+var cwd = process.cwd();
+var isModulePackage = getPackageType() === 'module';
+
+function getPackageType() {
```

Comment:
can we use https://npmjs.com/get-package-type for this?

Reply:
that package treats an empty type as commonjs, which i understood from you does not undo a parent package.json's `type: module`...

Reply:
The goal of `get-package-type` is to determine if node.js will interpret `/path/to/file.js` as a module or commonjs.  Any discrepancy is a bug which I would want to fix.

Reply:
awesome, it would be much nicer to use your library instead of my subpar solution!

Reply:
i just tested, it appears that a `package.json` without a "type" field does overwrite a parent "type: module". Ie if the closest `package.json` is missing the "type" field, the file is interpreted as commonjs.

Reply:
i just replaced my implementation with get-package-type... however i'm beginning to wonder if we should not just always use `import()` and let node determine esm vs cjs loading internally. @ljharb are there any downsides to always using `import()`?

Reply:
I suppose the only downside would be unnecessary asynchrony.

Reply:
Would this cause spurious warnings about ES modules being experimental on certain versions of node.js?

Reply:
Yes, but those could be suppressed.